enhancements/network/baremetal-ipi-network-configuration.md

```diff
@@ -0,0 +1,187 @@
+---
+title: baremetal-ipi-network-configuration
+authors:
+  - "@cybertron"
+  - "@hardys"
+reviewers:
+  - "@kirankt"
+  - "@dtantsur"
+  - "@zaneb"
+approvers:
+  - "TBD"
+creation-date: 2021-05-21
+last-updated: 2021-06-28
+status: provisional|implementable
+
+see-also:
+  - "/enhancements/host-network-configuration.md"
+  - "/enhancements/machine-config/mco-network-configuration.md"
+  - "/enhancements/machine-config/rhcos/static-networking-enhancements.md"
+
+# Baremetal IPI Network Configuration
+
+Describe user-facing API for day-1 network customizations in the IPI workflow,
+with particular focus on baremetal where such configuration is a common
+requirement.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently in the IPI flow, there is no way to provide day-1 network configuration
+which is a common requirement, particularly for baremetal users.  We can build
+on the [UPI static networking enhancements](rhcos/static-networking-enhancements.md)
+to enable such configuration in the IPI flow.
+
+## Motivation
+
+Since the introduction of baremetal IPI, a very common user request is how
+to configure day-1 networking, and in particular the following cases which are not currently possible:
+
+* Deploy with OpenShift Machine network on a tagged (non-default) VLAN
+* Deploy with OpenShift Machine network using static IPs (no DHCP)
+
+In both cases, this configuration cannot be achieved via DHCP so some
+means of providing the configuration to the OS is required.
+
+In the UPI flow this is achieved by consuming user-provided NetworkManager
+keyfiles, as an input to `coreos-install --copy-network`, but there is
+no corresponding user interface at the openshift-install level.
+
+### Goals
+
+* Define API for day-1 network customizations
+* Enable common on-premise network configurations (bond+vlan, static ips) via IPI
```

Comment:
Yeah I guess bond configuration could differ between hosts or groups of hosts (different nic names etc) - I think the API described here is basically always config per-host, but we can then perhaps optimize by detecting common config (to avoid duplicate images).

I also wonder if we have per-host config in the install-config.yaml if we can make yaml anchors/aliases work so that e.g in the case where you have a single bond/vlan config for every host you don't have to copy/paste it for each host.